src/script/descriptor.h

```diff
@@ -13,6 +13,18 @@
 
 #include <vector>
 
+// Cache for pre-computed descriptor things
+class DescriptorCache {
+private:
+    unsigned int m_pos_begin;
+    std::vector<std::vector<CPubKey>> m_pubkey_cache;
+
+public:
+    DescriptorCache() : m_pos_begin(0) {}
+    DescriptorCache(int begin_pos) : m_pos_begin(begin_pos) {}
+    bool CachePubKey(unsigned int cache_pos, const CPubKey& pubkey);
+    bool GetCachedPubKey(unsigned int global_pos, unsigned int internal_pos, CPubKey& pubkey) const;
```

Comment:
`global_pos` and `internal_pos` could use documentation. Would it make sense to cache all pubkeys (for a given position) in one operation?